applications/StructuralMechanicsApplication/custom_utilities/shell_cross_section.cpp

```diff
@@ -758,155 +714,95 @@ int ShellCrossSection::Check(const Properties& rMaterialProperties,
 {
     KRATOS_TRY
 
-    if(this->mEditingStack)
-        KRATOS_THROW_ERROR(std::logic_error, "The Ply Stack of a ShellCrossSection is in Editing mode", "")
+    KRATOS_ERROR_IF(this->mEditingStack) << "The Ply Stack of a ShellCrossSection is in Editing mode" << std::endl;
 
-        if(this->mStack.size() < 1)
-            KRATOS_THROW_ERROR(std::logic_error, "The Ply Stack of a ShellCrossSection cannot be empty", "")
+    if(this->mStack.size() < 1)
+        KRATOS_THROW_ERROR(std::logic_error, "The Ply Stack of a ShellCrossSection cannot be empty", "")
 
-            if(this->mThickness <= 0.0)
-                KRATOS_THROW_ERROR(std::logic_error, "The Thickness of a ShellCrossSection should be a positive real number", this->mThickness)
+    if(this->GetThickness() <= 0.0)
+        KRATOS_THROW_ERROR(std::logic_error, "The Thickness of a ShellCrossSection should be a positive real number", this->GetThickness())
 
-                for(PlyCollection::iterator ply_it = mStack.begin(); ply_it != mStack.end(); ++ply_it)
-                {
-                    Ply& iPly = *ply_it;
+    for (auto& r_ply : mStack)
+    {
+        Ply::IntegrationPointCollection::size_type numip( r_ply.NumberOfIntegrationPoints() );
+        if(numip < 1)
+            KRATOS_THROW_ERROR(std::logic_error, "The number of integration points in a Ply is not set properly", numip);
 
-                    Ply::IntegrationPointCollection::size_type numip( iPly.NumberOfIntegrationPoints() );
-                    if(numip < 1)
-                        KRATOS_THROW_ERROR(std::logic_error, "The number of integration points in a Ply is not set properly", numip);
+        if(r_ply.GetPropertiesPointer() == NULL)

```

Comment:
This was not added by me

I would have wanted to cahnge all the error macros in a separate PR to not pollute this one but I can also do it here then

Reply:
I understand not wanting to expand this PR, however I commented this because it appears to be added. Still I think the NULL and KRATOS_THROW_ERROR are doable in this PR, but I am not insisting on it.